vhost-device-console/src/tests.rs

```diff
@@ -0,0 +1,686 @@
+// VIRTIO CONSOLE Emulation via vhost-user
```

Comment:
I always appreciated having the tests close to the code, this also prevented us from making `pub` several fields used only by the tests.

What is the advantage of moving them to a separate module?

Reply:
Only advantage is not recompiling stuff when test code changes but crate code doesn't. I guess I was being overzealous, I will drop that commit :D